### PR TITLE
Editor: Fix duplicate save panels

### LINKED ISSUE #62702: [WP6.6] Publish Panel: Unexpected Behavior in the Site Editor
### Description

Originally reported on Slack: https://wordpress.slack.com/archives/C02QB2JS7/p1718858165581179

In the editor, you can use the tab key to focus on the "Open publish panel" button that appears in the bottom right. Clicking this button will show the Publish panel.

This is expected behavior in the post editor, but may cause unexpected behavior in the site editor.

This issue does not occur in WP6.5.

### Step-by-step reproduction instructions

- Access the site editor.
- Click the tab key repeatedly to move focus to the "Open puglish panel" button.
- Click that button.
- The Publish panel appears.
- When you hover the mouse over the editor canvas, the Publish panel moves inside the canvas.

The following issue occurs whether the canvas is in view or edit mode.

- Open any template.
- Click the tab key repeatedly to move focus to the "Open puglish panel" button.
- Click that button.
- The Publish panel appears.
- URL is undefined.
- Copy button doesn't work.
- The 'View Template' button doesn't work.
- Clicking the 'Add New Template' button shows an error.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/54422211/a2352eb5-ba1e-4699-b1f0-d7e297eeb542

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
Part of: https://github.com/WordPress/gutenberg/issues/62694
Related: https://github.com/WordPress/gutenberg/pull/62813
Also fixes: https://github.com/WordPress/gutenberg/issues/62702

This PR does a couple of things:
1. In site editor we would render two `save panels` and you could `tab` through both of them. Here I render only one and for now I followed the same approach we take for the `customSaveButton`. We pass the custom save panel only when we are previewing a theme and always render the site editor's save panel in sidebar and not in its layout component. This eventually should be handled in the `editor` package probably without needing to pass these extra custom save button and panel.

https://github.com/WordPress/gutenberg/assets/16275880/85df6563-d83f-4781-aa91-38a9030a927d

 

2. Currently the default save panel would be to `open publish panel` and here I changed it to `open save panel` which makes more sense in most cases, since an entity couldn't be `publishable` at all. This fixes [this comment](https://github.com/WordPress/gutenberg/issues/62702#issuecomment-2188441995) too.

https://github.com/WordPress/gutenberg/assets/16275880/bd8b477b-7f12-4d4a-87d2-e8d68e3e3a80

3. Added some extra checks to properly disable the default `open save panel` button.

## Testing Instructions
1. No regression for this PR: https://github.com/WordPress/gutenberg/pull/62813. When in `view` mode in site editor we shouldn't be able to navigate to a save panel using tab.
2. No rendering of two save panels. This can be tested in site editor.
3. The default save panel is `open save panel` and is disabled when needs to.
5. Previewing and activating themes in site editor works as before with any combination of other changes or not.
6. In site editor the `save button` in the left sidebar should work exactly as before and render the save changes modal and save properly

### CODE REVIEW

File: test/e2e/specs/editor/various/switch-to-draft.spec.js [hunk 1]
```diff
@@ -57,7 +57,9 @@ test.describe( 'Clicking "Switch to draft" on a published/scheduled post/page',
 							.getByRole( 'button', { name: 'Close Settings' } )
 							.click();
 					}
-					await page.getByRole( 'button', { name: 'Save' } ).click();
+					await page
+						.getByRole( 'button', { name: 'Save', exact: true } )
+						.click();
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/switch-to-draft.spec.js]: Suggestion (non-blocking): We can also add scope to the locator. Example #62454.

**Comment by Mamaduka** [comment]
@ntsekouras, follow-ing e2e test is still failing:

```
[chromium] › site-editor/multi-entity-saving.spec.js:29:2 › Site Editor - Multi-entity save flow › save flow should work as expected
```

**Comment by ntsekouras** [comment]
>@ntsekouras, follow-ing e2e test is still failing:

Yeah, I updated it.

File: packages/editor/src/components/editor-interface/index.js [hunk 2]
```diff
@@ -214,7 +215,7 @@ export default function EditorInterface( {
 				)
 			}
 			actions={
-				! isPreviewMode ? (
+				customSavePanel || (
```
Comment (youknowriad) [inline packages/editor/src/components/editor-interface/index.js]: I think we should keep isPreviewMode check here. I mean if you don't have customSavePanel and you're in preview mode, we shouldn't render the save panel.

File: packages/edit-site/src/components/layout/index.js [hunk 3]
```diff
@@ -236,8 +235,6 @@ export default function Layout( { route } ) {
 						</div>
 					) }
 				</div>
-
-				<SavePanel />
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/index.js:240]: This PR probably removes the save panel from pages that don't render an editor, I think that's probably a bad idea. We should be able to "save" from these pages as well.

**Comment by ntsekouras** [comment]
I had a chat with @youknowriad in private about some of the challenges this PR and the approach here, regarding his previous two comments. We agreed to preserve the `isPreviewMode` in `EditorInterface` that is effective for all editors (previously I had moved this check in site editor's editor) and render site editor's save panel in sidebar instead of its layout component.

File: packages/edit-site/src/components/layout/index.js (hunk 3 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/layout/index.js:240]: see: https://github.com/WordPress/gutenberg/pull/62863#issuecomment-2201990675

File: packages/editor/src/components/editor-interface/index.js (hunk 2 above)
Comment (ntsekouras) [reply packages/editor/src/components/editor-interface/index.js]: see: https://github.com/WordPress/gutenberg/pull/62863#issuecomment-2201990675

File: packages/editor/src/components/editor-interface/index.js [hunk 4]
```diff
@@ -214,18 +215,22 @@ export default function EditorInterface( {
...
-							entitiesSavedStatesCallback
-						}
-						setEntitiesSavedStatesCallback={
-							setEntitiesSavedStatesCallback
-						}
-						forceIsDirtyPublishPanel={ forceIsDirty }
-					/>
-				) : undefined
+				! isPreviewMode
+					? customSavePanel || (
```
Comment (youknowriad) [inline packages/editor/src/components/editor-interface/index.js:219]: I'm guessing these two conditions can be merged using a `&&`

File: test/e2e/specs/site-editor/multi-entity-saving.spec.js [hunk 5]
```diff
@@ -41,7 +41,7 @@ test.describe( 'Site Editor - Multi-entity save flow', () => {
 		).toBeEnabled();
 		await expect(
 			page
-				.getByRole( 'region', { name: 'Save panel' } )
+				.getByRole( 'region', { name: 'Editor publish' } )
```
Comment (youknowriad) [inline test/e2e/specs/site-editor/multi-entity-saving.spec.js:44]: Why is this change needed? The region is for saving not just publishing no?

**Review by youknowriad — APPROVED** [review]
Works well in my testing.

File: test/e2e/specs/site-editor/multi-entity-saving.spec.js (hunk 5 above)
Comment (ntsekouras) [reply test/e2e/specs/site-editor/multi-entity-saving.spec.js:44]: Yeah, but that's what we [call it ](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/editor-interface/index.js#L43)for a while.. Should we change it in a follow up?

File: packages/editor/src/components/editor-interface/index.js (hunk 4 above)
Comment (ntsekouras) [reply packages/editor/src/components/editor-interface/index.js:219]: They could, but it would be like ` !x && a || b` and because of the `||` I think it's a bit clearer this way..
